core/main.ts

```diff
@@ -102,3 +104,54 @@ export function main(coreExecutionRequest: Uint8Array | string): Uint8Array | st
 
   return dataform.CoreExecutionResponse.encode(coreExecutionResponse).finish();
 }
+
+function loadActionConfigs(session: Session, filePaths: string[]) {
+  filePaths
+    .filter(
+      path => path.startsWith(`definitions${utils.pathSeperator}`) && path.endsWith("actions.yaml")
+    )
+    .sort()
+    .forEach(actionConfigsPath => {
+      let actionConfigsAsJson = {};
+      try {
+        // tslint:disable-next-line: tsr-detect-non-literal-require
+        actionConfigsAsJson = nativeRequire(actionConfigsPath).asJson();
+      } catch (e) {
+        session.compileError(e, actionConfigsPath);
+      }
+      // TODO(ekrekr): Throw nice errors if the proto is invalid.
+      verifyObjectMatchesProto(dataform.ActionConfigs, actionConfigsAsJson);
+      const actionConfigs = dataform.ActionConfigs.fromObject(actionConfigsAsJson);
+
+      actionConfigs.actions.forEach(actionConfig => {
+        const { fileExtension, fileNameAsTargetName } = extractActionDetailsFromFileName(
+          actionConfig.fileName
+        );
+        if (!actionConfig.target?.name) {
+          if (!actionConfig.target) {
+            actionConfig.target = {};
+          }
+          actionConfig.target.name = fileNameAsTargetName;
+        }
```

Comment:
IMO this would be clearer as:

```
if (!actionConfig.target) {
  actionConfig.target = {};
}
if (!actionConfig.target.name) {
  actionConfig.target.name = fileNameAsTargetName;
}
```